crates/rpc/src/v03/method/simulate_transaction.rs

```diff
@@ -0,0 +1,495 @@
+                                    entry_point_type: Some(EntryPointType::Constructor),
+                                    events: Some(vec![]),
+                                    function_call: FunctionCall {
+                                        calldata: vec![
+                                            felt!("0x063C056DA088A767A6685EA0126F447681B5BCEFF5629789B70738BC26B5469D"),
+                                        ],
+                                        contract_address: Address(felt!("0x0332141F07B2081E840CD12F62FB161606A24D1D81D54549CD5FB2ED419DB415")),
+                                        entry_point_selector: felt!("0x028FFE4FF0F226A9107253E17A904099AA4F63A02A5621DE0576E5AA71BC5194"),
+                                    },
+                                    messages: Some(vec![]),
+                                    result: Some(vec![]),
+                                },
+                            ),
+                            validate_invocation: Some(
+                                FunctionInvocation {
+                                    call_type: Some(CallType::Call),
+                                    caller_address: Some(felt!("0x0")),
+                                    calls: Some(vec![]),
+                                    code_address: Some(felt!("0x02B63CAD399DD78EFBC9938631E74079CBF19C9C08828E820E7606F46B947513")),
+                                    entry_point_type: Some(EntryPointType::External),
+                                    events: Some(vec![]),
+                                    function_call: FunctionCall {
+                                        calldata: vec![
+                                            felt!("0x02B63CAD399DD78EFBC9938631E74079CBF19C9C08828E820E7606F46B947513"),
+                                            felt!("0x046C0D4ABF0192A788ACA261E58D7031576F7D8EA5229F452B0F23E691DD5971"),
+                                            felt!("0x063C056DA088A767A6685EA0126F447681B5BCEFF5629789B70738BC26B5469D"),
+                                        ],
+                                        contract_address: Address(felt!("0x0332141F07B2081E840CD12F62FB161606A24D1D81D54549CD5FB2ED419DB415")),
+                                        entry_point_selector: felt!("0x036FCBF06CD96843058359E1A75928BEACFAC10727DAB22A3972F0AF8AA92895"),
+                                    },
+                                    messages: Some(vec![]),
+                                    result: Some(vec![]),
+                                },
+                            ),
+                            fee_transfer_invocation: None,
+                        },
+                    ),
+                ),
+            }]
+        };
+
+        let result = simulate_transaction(rpc, input).await.expect("result");
+
+        if db_path.exists() {
+            std::fs::remove_file(&db_path).expect("clean up");
+        }
```

Comment:
There are two problems with this:

- Doesn't clean up `-wal` and `-shm` files (we're using SQLite in write-ahead-log mode). Maybe changing the `JournalMode` when creating the database could help? (Haven't tried that, but should.)
- Does no cleanup if any assertion/unwrap() fails before the test finishes.

Reply:
Good point, added small `tempdir` util, might be useful in other tests: https://github.com/eqlabs/pathfinder/pull/977/commits/0f04bbed7e805e15ffb2df1576736a542658a3ab